initial-configuration/install-dependencies-docker.sh

```diff
@@ -0,0 +1,175 @@
+  yum -y update
+  # This repo can be removed after we move away from centos 7 I think
+  yum-config-manager  --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+  yum install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+  sudo systemctl start docker
+fi
+
+if [ -n "$(command -v apt-get)" ] && [ -z "$(command -v docker)" ]; then
+  echo "Apt detected. Assuming Debian. Install commands will use apt"
+  # Add Docker's official GPG key:
+  apt-get update
+  apt-get install ca-certificates curl gnupg
+  install -m 0755 -d /etc/apt/keyrings
+  curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+  chmod a+r /etc/apt/keyrings/docker.gpg
+
+  # Add the repository to Apt sources:
+  echo \
+    "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+    "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+    tee /etc/apt/sources.list.d/docker.list > /dev/null
+  apt-get update
+
+  # Install docker
+  apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+fi
+
+if [ -z "$(command -v docker)" ]; then
+  echo "You dont have docker installed and we cant detect a supported package manager."
+  echo "Install docker and then rerun this script"
+  exit
+else
+  echo "Looks like docker is installed! Attempting a simple docker command: "
+fi
+
+if ! docker run hello-world > /dev/null 2>&1; then
+  echo "Docker hello-world failed. Exiting"
+  exit
+else
+  echo "Docker daemon seems happy!"
+fi
+
+echo "Creating docker networks for PIC-SURE and HPDS"
+if [ -z "$(docker network ls --format '{{.Name}}' | grep ^picsure$)" ]; then
+  docker network create picsure
+else
+  echo "picsure network already exists. Leaving it alone."
+fi
+if [ -z "$(docker network ls --format '{{.Name}}' | grep ^hpdsNet$)" ]; then
+  docker network create hpdsNet
+else
+  echo "hpdsNet network already exists. Leaving it alone."
+fi
+
+
+#-------------------------------------------------------------------------------------------------#
+#                                           MySQL Start                                           #
+#                     Install Jenkins and configure jobs and DB connection                        #
+#-------------------------------------------------------------------------------------------------#
+if [ -z "$(docker ps --format '{{.Names}}' | grep picsure-db)" ]; then
+  echo "Starting mysql server"
+  echo "` < /dev/urandom tr -dc @^=+$*%_A-Z-a-z-0-9 | head -c${1:-24}`%4cA" > pass.tmp
+  echo "PICSURE_DB_ROOT_PASS=`cat pass.tmp`" >> mysql-docker/.env
+  echo "PICSURE_DB_PASS=`cat pass.tmp`" >> mysql-docker/.env
+  echo "PICSURE_DB_DATABASE=ignore" >> mysql-docker/.env
+  echo "PICSURE_DB_USER=ignore" >> mysql-docker/.env
+  cd mysql-docker
+  docker-compose up -d
+
+  echo "Sleeping 60 secs so that mysql can spin up."
+  sleep 60
```

Comment:
Could we use the health check implemented in the docker-compose instead?

The health check from the compose:
```
test: [ "CMD", "mysqladmin" ,"ping", "-uroot", "-p$PICSURE_DB_ROOT_PASS" ]
```

It would be a lot more verbose to use the health check, but also more dependable.
```
echo "Waiting for MySQL to become healthy..."
SECONDS=0
TIMEOUT=180
while [ $SECONDS -lt $TIMEOUT ]; do
    HEALTH=$(docker inspect --format='{{.State.Health.Status}}' picsure-db)
    if [ "$HEALTH" = "healthy" ]; then
        echo "MySQL is up and healthy."
        break
    fi
    echo "Waiting for MySQL to become healthy..."
    sleep 10
done

if [ "$HEALTH" != "healthy" ]; then
    echo "MySQL did not become healthy within $TIMEOUT seconds."
    # Handle the error case as needed
fi
```

Reply:
since you did the work for me, yes